simple_use_case/snakemake/Snakefile

```diff
@@ -1,65 +1,117 @@
+configfile: "./config.yaml"
+
+
+# NOTE 
+# $ snakemake --use-conda --cores 1 --config domain_size=4.0
+# does not re-execute the job generated from rule 'generate_mesh'
+# if it was executed before with e.g. the default value 2.0
+# It seems one has to force the execution of 'somerule' in this case
+# $ snakemake -R somerule
 rule generate_mesh:
-	input:
-		"../source/unit_square.geo"
-	output:
-		"preprocessing/unit_square.msh"
-	conda:
-		"../source/envs/preprocessing.yaml"
-	shell:
-		"gmsh -2 -order 1 -format msh2 {input} -o {output}"
+    input:
+        "../source/unit_square.geo"
+    output:
+        msh="preprocessing/square.msh",
+        stdout="preprocessing/gmsh_stdout.txt"
+    conda:
+        "../source/envs/preprocessing.yaml"
+    shell:
+        "gmsh -2 -setnumber domain_size {config[domain_size]} {input} -o {output.msh} > {output.stdout}"
```

Comment:
I actually didn't expose domain_size as an option in the cwl implementation. I thought we had said that the geo file is the input, which spits out the domain size. So I may have to adjust that then...

Or should we simply say geofile & size come as arguments to the workflow, and then we call gmsh with domainsize to create a domain of the desired size, and we simply pass that input value down to the macro-preparation-job instead of reading it from gmsh?

I would say reading it from gmsh's output only makes sense if don't explicitly pass it in. The requirements the wf poses on suitable geo files is different in the two cases. We either expect the geo file be parameterized by a size, or we expect it to tell us the size of the domain that was created... I think both are valid scenario, I just wouldn't mix them.. Which one would you prefer?

Reply:
okay, I would then prefer to pass domain size to the gmsh job and simply pass it down instead of parsing the output. I thought it be nice to have this as a configuration parameter that you could pass on the command line when executing the workflow. 
The case of the non-file output is then still covered by the number of Dofs.

Reply:
By the way, should we define `degree` as a configuration parameter as well? It would make sense to me, but is also not that important ...

Reply:
> So I change cwl again to not read the size from the gmsh output but to receive it as input, actually passing it down to gmsh..

Thanks! 
I will also change this here accordingly. 
> I don't mind. I think the importance is to cover the different types of in/outputs and intermediate data transfer and I think cover that. We can add it or leave it, no strong opinion here. I'd leave that up to you.

I will not add degree as configuration parameter.